apps/meteor/client/views/room/components/body/composer/ComposerReadOnly.tsx

```diff
@@ -1,9 +1,43 @@
-import { MessageFooterCallout } from '@rocket.chat/ui-composer';
-import { useTranslation } from '@rocket.chat/ui-contexts';
+import { Button } from '@rocket.chat/fuselage';
+import { MessageFooterCallout, MessageFooterCalloutContent } from '@rocket.chat/ui-composer';
+import { useEndpoint, useTranslation } from '@rocket.chat/ui-contexts';
 import type { ReactElement } from 'react';
-import React from 'react';
+import React, { useCallback, useRef, useEffect } from 'react';
+
+import { useRoom, useUserIsSubscribed } from '../../../contexts/RoomContext';
 
 export const ComposerReadOnly = (): ReactElement => {
 	const t = useTranslation();
-	return <MessageFooterCallout>{t('room_is_read_only')}</MessageFooterCallout>;
+	const room = useRoom();
+	const isSubscribed = useUserIsSubscribed();
+	const calloutRef = useRef() as React.MutableRefObject<HTMLButtonElement>;
+	const joinEndpoint = useEndpoint('POST', '/v1/channels.join');
+
+	useEffect(() => {
+		calloutRef.current.style.flex = !isSubscribed ? '4' : 'none';
+	}, [isSubscribed]);
+
+	const join = useCallback(
+		async (_e) => {
+			try {
+				await joinEndpoint({ roomId: room._id });
+			} catch (error: any) {
+				console.log(error);
```

Comment:
```suggestion
			} catch (error: unknown) {
				dispatchToastMessage({ type: 'error', message: error });
```
Also fix the import;